### PR TITLE
style: Simplify style sharing code.

### PR DESCRIPTION
**Pull request by emilio**
No functional changes, just some code cleanup.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15888)

### CODE REVIEW

File: components/style/matching.rs [hunk 1]
```diff
@@ -881,8 +881,10 @@ pub trait MatchMethods : TElement {
         }
 
         let mut should_clear_cache = false;
-        for (i, &mut (ref mut candidate, ())) in style_sharing_candidate_cache.iter_mut().enumerate() {
-            let sharing_result = self.share_style_with_candidate_if_possible(shared_context, candidate);
+        for (i, ref mut candidate) in style_sharing_candidate_cache.iter_mut().enumerate() {
```
Comment (Ms2ger) [inline components/style/matching.rs]: Should not need the `ref mut` anymore

**Review by Ms2ger — COMMENTED** [review]
r=me on the first commit with that change; I can't evaluate the second.

**Comment by emilio** [comment]
Whoops, the second commit was supposed to go on a different PR. I dropped it.

@bors-servo r=Ms2ger

**Comment by emilio** [comment]
@bors-servo r=Ms2ger

 * Removed unit tests using unexisting APIs.

**Comment by KiChjang** [comment]
@bors-servo retry
* Exception?!

**Pull request merged**